src/components/Claim/useClaimLink.tsx

```diff
@@ -79,19 +80,16 @@ export const useClaimLink = () => {
     }) => {}
 
     const switchNetwork = async (chainId: string) => {
-        if (currentChain?.id.toString() !== chainId.toString()) {
-            setLoadingState('Allow network switch')
-
-            try {
-                await switchChainAsync({ chainId: Number(chainId) })
-                setLoadingState('Switching network')
-                await new Promise((resolve) => setTimeout(resolve, 2000))
-                setLoadingState('Loading')
-            } catch (error) {
-                setLoadingState('Idle')
-                console.error('Error switching network:', error)
-                // TODO: handle error, either throw or return error
-            }
+        try {
+            await switchNetworkUtil({
+                chainId,
+                currentChainId: currentChain?.id,
+                setLoadingState,
+                switchChainAsync,
+            })
+            console.log(`Switched to chain ${chainId}`)
+        } catch (error) {
+            console.error('Failed to switch network:', error)
```

Comment:
idg this - why are you wrapping switchNetworkUtil in switchNetwork and adding an error log? whats the point?